AttachmentElement uses AppKit in WebContent process

kkinnunen-apple opened the pull request:
#### 62439d6ff75ce918cff1584fde18e920eaac4901
<pre>
AttachmentElement uses AppKit in WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=262684">https://bugs.webkit.org/show_bug.cgi?id=262684</a>
rdar://116509223

Reviewed by Aditya Keerthi.

* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::paintAttachment):

Attachment painting is implemented using GraphicsContext.
It should not set the AppKit NSGraphicsContext. If the rendering mode
is GPUP, the LocalCurrentGraphicsContext is no-op since the
context does not have a platform context.

* Source/WebCore/platform/graphics/cocoa/GraphicsContextCocoa.mm:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::ControlMac::drawListButton):
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::drawMenuListBackground):
* Source/WebCore/platform/graphics/mac/controls/ProgressBarMac.mm:
(WebCore::ProgressBarMac::draw):

Remove use of ContextContainer, it is no-op code.
Replace with direct GraphicsContext::platformContext() call.

* Source/WebCore/platform/graphics/mac/IconMac.mm:
* Source/WebCore/platform/mac/ThemeMac.mm:

Remove unused LocalCurrentGraphicsContext.h include.

Canonical link: <a href="https://commits.webkit.org/268978@main">https://commits.webkit.org/268978@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/adf6e2230514c50718d8159ed1a545722cd2ba3a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21021 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21402 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22087 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22902 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19556 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21261 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24655 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21587 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20823 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21244 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20980 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18232 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23756 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18140 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19060 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25337 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19220 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19261 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23266 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19805 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16829 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19075 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5081 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23377 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19649 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Nit: I would retitle the PR to "AttachmentElement attempts to use platform graphics contexts in the WebContent process" or something similar.

There is nothing wrong with using AppKit itself in the WebContent process.

kkinnunen-apple committed:
AttachmentElement uses AppKit in WebContent process
https://bugs.webkit.org/show_bug.cgi?id=262684
rdar://116509223

Reviewed by Aditya Keerthi.

* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::paintAttachment):

Attachment painting is implemented using GraphicsContext.
It should not set the AppKit NSGraphicsContext. If the rendering mode
is GPUP, the LocalCurrentGraphicsContext is no-op since the
context does not have a platform context.

* Source/WebCore/platform/graphics/cocoa/GraphicsContextCocoa.mm:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::ControlMac::drawListButton):
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::drawMenuListBackground):
* Source/WebCore/platform/graphics/mac/controls/ProgressBarMac.mm:
(WebCore::ProgressBarMac::draw):

Remove use of ContextContainer, it is no-op code.
Replace with direct GraphicsContext::platformContext() call.

* Source/WebCore/platform/graphics/mac/IconMac.mm:
* Source/WebCore/platform/mac/ThemeMac.mm:

Remove unused LocalCurrentGraphicsContext.h include.

Canonical link: https://commits.webkit.org/268978@main
